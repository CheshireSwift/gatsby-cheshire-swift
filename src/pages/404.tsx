import * as React from 'react';
import { css } from '../../node_modules/emotion';

const NotFoundPage = () => (
  <div>
    <h1
      className={css({
        color: 'inherit',
        fontFamily:
          'Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif',
        fontSize: '2em',
        fontWeight: 'bold',
        lineHeight: 1.1,
        margin: 0,
        marginBottom: '1.45rem',
        marginTop: '1.45rem',
        padding: 0,
      })}
    >
      NOT FOUND
    </h1>
    <p
      className={css({
        "margin": 0,
        'margin-bottom': '1.45rem',
        "padding": 0,
      })}
    >
      You just hit a route that doesn&#39;t exist... the sadness.
    </p>
  </div>
);

export default NotFoundPage;
