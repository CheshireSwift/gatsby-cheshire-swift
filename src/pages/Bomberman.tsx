import * as React from 'react';
import * as BombermanGame from '../components/Bomberman';

export default () => (
  <div>
    <BombermanGame.BombermanGame />
  </div>
);
