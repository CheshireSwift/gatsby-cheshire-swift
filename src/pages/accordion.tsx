import * as React from 'react';
import Link from 'gatsby-link';
import Accordion from '../components/accordionDemo';
import { css } from 'emotion';

const myContent = [
  { title: 'title 1', content: 'text 1' },
  { title: 'title 2', content: 'text 2' },
  { title: 'title 3', content: 'text 3' },
];

const AccordionPage = () => (
  <div className={css({ fontFamily: 'Arial' })}>
    <h1>Accordion Demo</h1>
    <br />
    <Accordion content={myContent} />
    <br />
    <Link to="/">Go back to the homepage</Link>
  </div>
);

export default AccordionPage;
