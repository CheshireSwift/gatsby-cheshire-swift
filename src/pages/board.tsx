import * as React from 'react';
import * as _ from 'lodash';
import * as life from './game-of-life';
import * as cell from './cellState';
import TimeControl from './timecontrol';
import Square from './square';

interface BoardState {
    squares: cell.state[];
    isRunning: boolean;
    timerToken: Timer;
}

export class Board extends React.Component<{}, BoardState> {
    sideLength: number;
    constructor(props: {}) {
        super(props);
        this.sideLength = 50;
        this.state = {
            isRunning: false,
            squares: Array(this.sideLength * this.sideLength).fill(cell.state.DEAD),
            timerToken: 0,
        };
    }

    render() {
        return (
            <div>
                <TimeControl onClickHandler={this.alternateAnimation.bind(this)} value={this.buttonText()} />
                <div className="separator"></div>
                {_.range(this.sideLength).map((row: number) => this.makeRowOfSquares(row))}
            </div>
        );
    }

    alternateAnimation() {
        if (!this.state.isRunning) {
            const id = setInterval(this.evolve.bind(this), 500);
            this.setState({ timerToken: id });
        } else {
            clearInterval(this.state.timerToken);
        }
        this.setState({ isRunning: !this.state.isRunning });
    }

    buttonText() {
        return (this.state.isRunning ? 'Stop' : 'Go');
    }

    evolve() {
        this.setState({ squares: life.computeNextGeneration(this.state.squares) });
    }

    makeRowOfSquares(i: number) {
        return (<div className="board-row">
            {_.range(i * this.sideLength, (i + 1) * this.sideLength).map((j: number) => this.renderSquare(j))}
        </div>
        );
    }

    renderSquare(i: number) {
        return (
            <Square
                key={i}
                styleClass={cell.style[this.state.squares[i]]}
                onClickHandler={() => this.createAliveCell(i)}
            />
        );
    }

    createAliveCell(i: number) {
        const squares = this.state.squares.slice();
        squares[i] = cell.state.ALIVE;
        this.setState({ squares });
    }

}
