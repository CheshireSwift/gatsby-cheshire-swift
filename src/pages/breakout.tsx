import * as React from 'react';
import Breakout from '../components/breakout';

const BreakoutPage = () => <Breakout />;

export default BreakoutPage;
