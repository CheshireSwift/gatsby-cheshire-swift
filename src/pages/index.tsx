import * as React from 'react';
import Link from 'gatsby-link';

export default () => (
  <div>
    <h1>Hi people</h1>
    <Link to="/page-2/">Go to page 2</Link>
    <br />
    <Link to="/ultimate/">Play Ultimate Tic-Tac-Toe</Link>
  </div>
);
