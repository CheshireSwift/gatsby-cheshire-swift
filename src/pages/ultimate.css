.square {
  background: #fff;
  border: 1px solid #999;
  line-height: 34px;
  height: 34px;
  margin-right: -1px;
  margin-top: -1px;
  padding: 0;
  text-align: center;
  width: 34px;
}

.small-board {
  position: relative;
  display: grid;
  grid-template-columns: repeat(3, minmax(auto, max-content));
  background-color: white;
  padding: 10px;
  border: 1px solid black;
}

.large-board {
  display: grid;
  grid-template-columns: repeat(3, minmax(auto, max-content));
  background-color: black;
  border: 1px solid black;
}

.board-available {
  background-color: green;
}

.board-unavailable {
  background-color: red;
}

.small-winner {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}
