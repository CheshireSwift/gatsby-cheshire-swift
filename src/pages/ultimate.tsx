import * as React from 'react';
import * as _ from 'lodash';
import { css } from 'emotion';

import { Nought, Cross } from '../components/shapes';

export enum Winner {
  Nought,
  Cross,
  Draw,
}

export interface Cell {
  player: Nullable<Winner>;
}

export interface Board {
  cells: Cell[];
}

function calculateWinner(squares: Cell[]): Nullable<Winner> {
  const lines = [
    [0, 1, 2],
    [3, 4, 5],
    [6, 7, 8],
    [0, 3, 6],
    [1, 4, 7],
    [2, 5, 8],
    [0, 4, 8],
    [2, 4, 6],
  ];
  const winningLine = lines.find(
    ([a, b, c]) =>
      squares[a].player !== null &&
      squares[a].player !== Winner.Draw &&
      squares[a].player === squares[b].player &&
      squares[a].player === squares[c].player,
  );
  if (winningLine) {
    return squares[winningLine[0]].player;
  } else if (squares.every(child => child.player !== null)) {
    return Winner.Draw;
  } else {
    return null;
  }
}

export interface GameState {
  available: number[];
  boards: Board[];
  xIsNext: boolean;
}

export class Game extends React.Component<{}, GameState> {
  constructor(props: {}) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
    this.state = {
      available: _.range(9),
      boards: _.times(9, i => ({
        cells: _.times(9, j => ({ player: null })),
      })),
      xIsNext: true,
    };
  }

  handleClick(i: number, j: number) {
    const boards = [...this.state.boards];
    boards[i] = {
      cells: [...boards[i].cells],
    };
    const smallBoardWinner = calculateWinner(boards[i].cells);
    if (smallBoardWinner || boards[i].cells[j].player !== null) {
      return;
    }
    boards[i].cells[j].player = this.state.xIsNext
      ? Winner.Cross
      : Winner.Nought;
    let available: number[];
    const overallWinner = calculateWinner(
      boards.map(board => ({ player: calculateWinner(board.cells) })),
    );
    if (overallWinner) {
      available = [];
    } else {
      const boardWinner = (board: Board) => calculateWinner(board.cells);
      available = boardWinner(boards[j])
        ? _.range(9).filter(board => !boardWinner(boards[board]))
        : [j];
    }
    this.setState(prevState => ({
      available,
      boards,
      xIsNext: !prevState.xIsNext,
    }));
  }

  render() {
    return (
      <div className={css({ textAlign: 'center' })}>
        <div className={css({ display: 'inline-block' })}>
          <LargeBoard
            available={this.state.available}
            boards={this.state.boards}
            handleClick={this.handleClick}
          />
        </div>
      </div>
    );
  }
}

export interface LargeBoardProps {
  available: number[];
  boards: Board[];
  handleClick: (i: number, j: number) => void;
}

const boardStyle = {
  display: 'grid',
  gridTemplateColumns: 'repeat(3, minmax(auto, max-content))',
};

export const LargeBoard = (props: LargeBoardProps) => (
  <div
    className={css({
      backgroundColor: 'black',
      border: '1px solid black',
      gridGap: 1,
      ...boardStyle,
    })}
  >
    {props.boards.map((board: Board, i: number) => {
      const availableBoard = props.available.includes(i);
      return (
        <SmallBoard
          key={i}
          available={availableBoard}
          board={board}
          handleClick={j => {
            if (availableBoard) {
              props.handleClick(i, j);
            }
          }}
        />
      );
    })}
  </div>
);

export interface SmallBoardProps {
  available: boolean;
  board: Board;
  handleClick: (i: number) => void;
}

export const SmallBoard = (props: SmallBoardProps) => {
  const winner = calculateWinner(props.board.cells);
  return (
    <div
      className={css({
        backgroundColor: props.available ? 'green' : 'red',
        padding: 10,
        position: 'relative',
        ...boardStyle,
      })}
    >
      {winner && (
        <div
          className={css({
            height: '100%',
            left: 0,
            position: 'absolute',
            top: 0,
            width: '100%',
          })}
        >
          {getSymbol(winner)()}
        </div>
      )}
      {props.board.cells.map((cell: Cell, i: number) => (
        <Square
          key={i}
          cell={cell}
          onClick={props.available ? () => props.handleClick(i) : null}
        />
      ))}
    </div>
  );
};

export interface SquareProps {
  cell: Cell;
  onClick: () => void;
}

export const Square = (props: SquareProps) => (
  <button
    className={css({
      backgroundColor: 'white',
      border: '1px solid #999',
      height: 34,
      marginRight: -1,
      marginTop: -1,
      padding: 0,
      width: 34,
    })}
    onClick={props.onClick}
  >
    {getSymbol(props.cell.player)()}
  </button>
);

function getSymbol(player: Winner): () => JSX.Element {
  switch (player) {
    case Winner.Nought:
      return () => <Nought />;
    case Winner.Cross:
      return () => <Cross />;
    default:
      return () => null;
  }
}

export default Game;
