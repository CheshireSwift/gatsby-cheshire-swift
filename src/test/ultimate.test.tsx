import * as React from 'react';
import { render, shallow, mount } from 'enzyme';
import * as _ from 'lodash';

import * as ultimate from '../pages/ultimate';

describe('ultimate tic-tac-toe', () => {
  it('Nought matches the snaphot', () => {
    expect(render(<ultimate.Nought />)).toMatchSnapshot();
  });

  it('Cross matches the snaphot', () => {
    expect(render(<ultimate.Cross />)).toMatchSnapshot();
  });

  it('Square renders nothing with no winner', () => {
    const component = render(
      <ultimate.Square cell={{ winner: null }} onClick={() => null} />,
    ).get(0);
    expect(component.children).toEqual([]);
  });

  it('Square renders winner', () => {
    const component = render(
      <ultimate.Square
        cell={{ winner: ultimate.Winner.Cross }}
        onClick={() => null}
      />,
    );
    expect(component.find('svg').length).toBeGreaterThanOrEqual(1);
  });

  it('Square calls onClick when clicked', () => {
    const onClick = jest.fn();
    const component = shallow(
      <ultimate.Square cell={{ winner: null }} onClick={onClick} />,
    );
    component.simulate('click');
    expect(onClick).toHaveBeenCalled();
  });

  it('SmallBoard has 9 Squares', () => {
    const component = render(
      <ultimate.SmallBoard
        available={true}
        board={{
          cells: Array<ultimate.Cell>(9).fill({ winner: null }),
          winner: null,
        }}
        handleClick={() => null}
      />,
    );
    expect(component.find('button').length).toBe(9);
  });

  it('SmallBoard accepts clicks if available', () => {
    const onClick = jest.fn();
    const component = shallow(
      <ultimate.SmallBoard
        available={true}
        board={{
          cells: Array<ultimate.Cell>(9).fill({ winner: null }),
          winner: null,
        }}
        handleClick={onClick}
      />,
    );
    component.childAt(0).simulate('click');
    expect(onClick).toHaveBeenCalledWith(0);
  });

  it('SmallBoard does not accepts clicks if not available', () => {
    const onClick = jest.fn();
    const component = shallow(
      <ultimate.SmallBoard
        available={false}
        board={{
          cells: Array<ultimate.Cell>(9).fill({ winner: null }),
          winner: null,
        }}
        handleClick={onClick}
      />,
    );
    component.childAt(0).simulate('click');
    expect(onClick).not.toHaveBeenCalled();
  });

  it('SmallBoard displays squares with the correct winners', () => {
    const component = shallow(
      <ultimate.SmallBoard
        available={false}
        board={{
          cells: [
            { winner: ultimate.Winner.Cross },
            { winner: ultimate.Winner.Nought },
            ...Array(7).fill({ winner: null }),
          ],
          winner: null,
        }}
        handleClick={() => null}
      />,
    );
    expect(
      component.children().map(child => child.prop('cell').winner),
    ).toEqual([
      ultimate.Winner.Cross,
      ultimate.Winner.Nought,
      ...Array(7).fill(null),
    ]);
  });

  it('SmallBoard displays a winner', () => {
    const component = render(
      <ultimate.SmallBoard
        available={false}
        board={{
          cells: Array(9).fill({ winner: null }),
          winner: ultimate.Winner.Nought,
        }}
        handleClick={() => null}
      />,
    );
    expect(component.find('svg').length).toBe(1);
  });

  it('LargeBoard has 9 SmallBoards', () => {
    const component = render(
      <ultimate.LargeBoard
        available={[1, 2, 3, 4, 5, 6, 7, 8, 0]}
        boards={Array(9).fill({
          cells: Array(9).fill({ winner: null }),
          winner: null,
        })}
        handleClick={() => null}
      />,
    );
    expect(component.find('div').length).toBe(9);
  });

  it('LargeBoard passes click handler', () => {
    const handleClick = jest.fn();
    const component = shallow(
      <ultimate.LargeBoard
        available={[1, 2, 3, 4, 5, 6, 7, 8, 0]}
        boards={Array(9).fill({
          cells: Array(9).fill({ winner: null }),
          winner: null,
        })}
        handleClick={handleClick}
      />,
    );
    component.childAt(1).prop('handleClick')(20);
    expect(handleClick).toHaveBeenCalledWith(1, 20);
  });
});
