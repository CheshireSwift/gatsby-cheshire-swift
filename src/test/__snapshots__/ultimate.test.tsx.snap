// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ultimate tic-tac-toe Cross matches the snaphot 1`] = `
<svg
  baseProfile="full"
  height="100%"
  version="1.1"
  width="100%"
>
  <line
    stroke="black"
    stroke-width="10%"
    x1="20%"
    x2="80%"
    y1="20%"
    y2="80%"
  />
  <line
    stroke="black"
    stroke-width="10%"
    x1="20%"
    x2="80%"
    y1="80%"
    y2="20%"
  />
</svg>
`;

exports[`ultimate tic-tac-toe Nought matches the snaphot 1`] = `
<svg
  baseProfile="full"
  height="100%"
  version="1.1"
  width="100%"
>
  <circle
    cx="50%"
    cy="50%"
    fill="transparent"
    r="30%"
    stroke="black"
    stroke-width="10%"
  />
</svg>
`;
